### PR TITLE
Added Link method to IUrlHelper

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #1966 @yishaigalatzer @pranavkm

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/IUrlHelper.cs [hunk 1]
```diff
@@ -55,5 +55,13 @@ public interface IUrlHelper
         /// <param name="routeContext">The context object for the generated URLs for a route.</param>
         /// <returns>The fully qualified or absolute URL.</returns>
         string RouteUrl([NotNull] UrlRouteContext routeContext);
+
+        /// <summary>
+        /// Generates an absolute URL using the specified route name and values.
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/IUrlHelper.cs]: Also mention that the host and scheme come from the current request

File: src/Microsoft.AspNet.Mvc.Core/UrlHelper.cs [hunk 2]
```diff
@@ -127,6 +127,19 @@ public virtual string Content([NotNull] string contentPath)
...
+        /// <inheritdoc />
+        public virtual string Link(string routeName, object values)
+        {
+            return RouteUrl(new UrlRouteContext()
+            {
+                RouteName = routeName,
+                Values = values,
+                Protocol = _httpContext.Request.Scheme,
+                Host = _httpContext.Request.Host.ToUriComponent(),
+                Fragment = null
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/UrlHelper.cs]: no need to initialize this. That's the whole point of the context

File: src/Microsoft.AspNet.Mvc.Core/IUrlHelper.cs [hunk 3]
```diff
@@ -55,5 +55,13 @@ public interface IUrlHelper
...
+        /// </summary>
+        /// <param name="routeName">The name of the route that is used to generate the URL.</param>
+        /// <param name="values">An object that contains the route values.</param>
+        /// <returns>The generated absolute URL.</returns>
+        string Link(string routeName, object values);
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/IUrlHelper.cs]: This should just be an extension method.

File: test/Microsoft.AspNet.Mvc.Core.Test/UrlHelperTest.cs [hunk 4]
```diff
@@ -728,6 +728,61 @@ public void UrlActionWithAllParameters_ReturnsExpectedResult()
...
+            var services = GetServices();
+            var routeCollection = GetRouter(services, "MyRouteName", "any/url");
+            var urlHelper = CreateUrlHelper("/app", routeCollection);
+
+            // Act
+            var url = urlHelper.Link("MyRouteName", null);
+
+            // Assert
+            Assert.Equal("http://localhost/app/any/url", url);
+        }
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/UrlHelperTest.cs]: include one where the scheme is non-default

**Comment by rynowak** [comment]
:watch:

File: test/Microsoft.AspNet.Mvc.Core.Test/UrlHelperTest.cs (hunk 4 above)
Comment (ajaybhargavb) [inline test/Microsoft.AspNet.Mvc.Core.Test/UrlHelperTest.cs]: Added.

**Comment by ajaybhargavb** [comment]
Updated.

File: src/Microsoft.AspNet.Mvc.Core/UrlHelper.cs (hunk 2 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/UrlHelper.cs]: Please delete this line, it's not needed.

**Comment by rynowak** [comment]
:shipit: after removing redundant init in `Link`

**Pull request merged**